### ISSUE TITLE
AWS SageMaker Scheduler

### ISSUE THREAD

**Issue by clumsy** [pytorch/torchx#785]
## Description
A new scheduler to run AWS SageMaker Training Jobs.

## Motivation/Background
Unlike AWS Batch that requires to use managing the fleet (within the current account), AWS SageMaker is providing a fleet of managed instances.

## Detailed Proposal
Add a new scheduler that triggers that takes care of preparing and triggering a new AWS SageMaker Training Job.

## Alternatives
N/A

## Additional context/links
https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-training.html

**Comment by clumsy** [comment]
I'm willing to contribute this one, but I kind of need https://github.com/pytorch/torchx/pull/781 to be addressed first. E.g. I would like to reuse getting EC2 instance type via `AwsResource`. @kiukchung

**Comment by kiukchung** [comment]
thanks, commented on #781 .

**Comment by kiukchung** [comment]
I've merged #781. Let me know if you are blocked on anything.
